training/src/anemoi/training/schemas/diagnostics.py

```diff
@@ -291,6 +297,55 @@ def clean_entity(cls: type["MlflowSchema"], values: dict[str, Any]) -> dict[str,
         return values
 
 
+class AzureMlflowSchema(BaseModel):
```

Comment:
why not inheriting from `MlflowSchema` and reusing the keys? we could also have a BaseMlflowSchema for common keys and then MlflowSchema and AzureMlflowSchema if ieasier

Reply:
We could totally do that. I made a separate schema because the options: offline, authentication, and save_dir don't make sense for azureml (in our first draft we just had an assertion error raise if offline was set to True). But we could bring back the assertion error and inherit, let me know what you think.

Just to give us something concrete, I'll make a BaseMLflowSchema for common options. But I'm happy to update that as needed.

Reply:
One downside of doing it this way, rather than having the azure version inherit options, is that the yaml situation gets a bit more complicated for users. For me to use the azure _target_, I can no longer modify all options from a top level config.yaml file, and instead I have to go into the diagnostics/evaluation.yaml file and remove the offline, authentication, and save_dir options. 

So, maybe it is better to just inherit the same schema, even though those options would get ignored. I'm not sure if you have a recommendation there.

Reply:
Thanks @anaprietonem! I added the `Annotated[ ]` around this, and made sure it works for me ([8a017bf](https://github.com/ecmwf/anemoi-core/pull/469/commits/8a017bf7f0289176d44779164422465e7d934d7a)).

That's a fair point regarding schema/class structure. Would you prefer it if I make the class structure consistent with the Schema structure? i.e. get rid of the BaseAnemoiMLflowLogger?